File: docs/03-user-guide/examples/python/grpc_execute_sql.py
#!/usr/bin/env python3
# gRPC: ExecuteSql example (requires generated proximadb.v1 stubs)

import grpc
from proximadb_v1_pb2 import ExecuteSqlRequest, SqlValue
from proximadb_v1_pb2_grpc import SqlServiceStub

def main():
    channel = grpc.insecure_channel("localhost:5679")
    stub = SqlServiceStub(channel)

    # SELECT with parameterized vector
    vec = SqlValue(value=SqlValue.Value(number_value=0.1))
    req = ExecuteSqlRequest(
        query="SELECT id FROM my_collection ORDER BY COSINE_DISTANCE(vector, $1) LIMIT 5",
        parameters=[SqlValue(value=SqlValue.Value(array_value=SqlValue.Array(items=[vec])))],
        collection="my_collection",
    )
    resp = stub.ExecuteSql(req)
    print(resp)

if __name__ == "__main__":
    main()

File: docs/03-user-guide/examples/rust/grpc_execute_sql.rs
// gRPC: ExecuteSql via tonic (requires generated client stubs)
// cargo add tonic prost tokio --features tokio/full

use proximadb_v1::sql_service_client::SqlServiceClient;
use proximadb_v1::{ExecuteSqlRequest, SqlValue};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = SqlServiceClient::connect("http://localhost:5679").await?;
    let vec = SqlValue { value: Some(proximadb_v1::sql_value::Value::NumberValue(0.1)) };
    let req = ExecuteSqlRequest {
        query: "SELECT id FROM my_collection ORDER BY COSINE_DISTANCE(vector, $1) LIMIT 5".to_string(),
        parameters: vec![SqlValue { value: Some(proximadb_v1::sql_value::Value::ArrayValue(proximadb_v1::SqlArray{ items: vec![vec] })) }],
        collection: Some("my_collection".to_string()),
        limit: None,
        offset: None,
    };
    let resp = client.execute_sql(req).await?;
    println!("{:?}", resp.into_inner());
    Ok(())
}

pub mod proximadb_v1 { include!("../../../../src/proto/proximadb.v1.rs"); }

